File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const map = new Map();
        const rec = [];
        const result = [];

        for (let i = 0; i < nums.length; i++) {
            const total = map.has(nums[i]) ? map.get(nums[i]) + 1 : 1;
            map.set(nums[i], total);
        }

        for (let [key, value] of map.entries()) {
            if (rec[value]) {
                rec[value] = [...rec[value], key];
            } else {
                rec[value] = [key];
            }
        }

        console.log(rec);

        for (let i = rec.length - 1; i >= 0; i--) {
            if (rec[i] != undefined && result.length < k) {
                result.push(...rec[i]);
            }
        }

        return result;
    }
}

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-5.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        result = [nums[0]];
        for i in range(len(nums)):
            if nums[i] != result[-1]:
                result.append(nums[i]) 

        for i in range(len(result)):
            nums[i] = result[i]
        return len(result)
        